pkg/metrics/metrics.go

```diff
@@ -343,18 +208,42 @@ func (c *Config) ensureServiceMonitor() error {
 						{
 							Action:      "replace",
 							Regex:       "(.*)",
-							Replacement: &replacement,
+							Replacement: &instanceLabel,
 							TargetLabel: "instance",
 							SourceLabels: []monv1.LabelName{
 								"__meta_kubernetes_endpoint_address_target_name",
 							},
 						},
+						{
```

Comment:
There is a lot of 'magic' happening here, requiring maintainers to be familiar with these actions and what they may be doing. Please add a short comment for each explaining what they are doing and why they need to be done, if it is not too obvious.